mlx/backend/metal/device.cpp

```diff
@@ -26,7 +26,8 @@ static constexpr const char* default_mtllib_path = METAL_PATH;
 
 auto load_device() {
   auto devices = MTL::CopyAllDevices();
-  auto device = static_cast<MTL::Device*>(devices->object(0));
+  auto device = static_cast<MTL::Device*>(devices->object(0))
+      ?: MTL::CreateSystemDefaultDevice();
```

Comment:
The copyAllDevices is not available on iOS so use a fallback:

- https://developer.apple.com/documentation/metal/1433367-mtlcopyalldevices